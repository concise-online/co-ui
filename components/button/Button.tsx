import React, { ButtonHTMLAttributes } from 'react';
import classNames from 'classnames';
import { getPrefixCls } from '../utils/config';

const prefixCls = getPrefixCls('btn');

interface Props extends Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'type'> {
  type?: 'primary' | 'success' | 'info' | 'warning' | 'danger';
  htmlType?: 'button' | 'submit' | 'reset';
  fill?: boolean;
}

function Button(props: Props) {
  const { children, className, htmlType, type, fill, ...btnProps } = props;
  const classes = classNames(prefixCls, className, {
    [`${prefixCls}-${type}`]: !!type,
    [`${prefixCls}-fill`]: fill,
  });

  return (
    <button
      type={htmlType}
      className={classes}
      {...btnProps}
    >
      <span>{children}</span>
    </button>
  );
}

export default Button;
