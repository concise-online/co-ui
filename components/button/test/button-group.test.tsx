import React from 'react';
import { shallow, render } from 'enzyme';
import Button from '../index';

describe('button group feature', () => {
  it('renders correctly', () => {
    const wrapper = render((
      <Button.Group>
        <Button>Prev</Button>
        <Button>Next</Button>
      </Button.Group>
    ));

    expect(wrapper).toMatchSnapshot();
  });

  it('the type of child can only be button', () => {
    const wrapper = shallow((
      <Button.Group>
        <Button>Prev</Button>
        <Button>Next</Button>
        <div>this is div node</div>
      </Button.Group>
    ));
    const children = wrapper.children();

    expect(children.someWhere((child) => child.is(Button))).toBeTruthy();
    expect(children).toHaveLength(2);
  });
});
