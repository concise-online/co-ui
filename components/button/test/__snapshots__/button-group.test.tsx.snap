// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button group feature renders correctly 1`] = `
<div
  class="co-btn-group"
>
  <button
    class="co-btn"
  >
    <span>
      Prev
    </span>
  </button>
  <button
    class="co-btn"
  >
    <span>
      Next
    </span>
  </button>
</div>
`;
